### PR TITLE
Update C++ headers - Follow up to #9282

### PR DESCRIPTION
**Pull request by SSoulaimane**
I forgot to update the C++ headers.

### CODE REVIEW

File: src/dmd/declaration.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-
+f
 /* Compiler implementation of the D programming language
```
Comment (thewilsonator) [inline src/dmd/declaration.h]: ???

**Review by thewilsonator — APPROVED** [review]
Otherwise looks good

File: src/dmd/declaration.h (hunk 1 above)
Comment (SSoulaimane) [reply src/dmd/declaration.h]: Sorry, typo.

**Comment by ibuclaw** [comment]
Where's the tests?

**Comment by SSoulaimane** [comment]
How?

**Comment by ibuclaw** [comment]
https://github.com/dlang/dmd/blob/master/src/tests/cxxfrontend.c

**Comment by ibuclaw** [comment]
To better catch run-time corruption, I think:
- Compile a nested function, and assert(fd->isNested())`
- Compile a non-nested function and `assert(!fd->isNested())`
- Compile a struct with an allocator, but no deallocator, and `assert(sd->aggNew && !sd->aggDelete)`

**Comment by SSoulaimane** [comment]
Ok, but why is checking `isNested()` related to the ABI?

**Comment by SSoulaimane** [comment]
Since the allocators were deprecated, I decided to check the next field (`ctor`) #10071.

**Pull request merged**